playbook/runner/runner.go

```diff
@@ -178,6 +178,70 @@ func getEligibleAgents(spec v1.PlaybookSpec, action *v1.PlaybookAction, run mode
 	return []string{Main}
 }
 
+func saveAIResultToSendHistory(ctx context.Context, run models.PlaybookRun) error {
+	if run.NotificationSendID == nil {
+		return nil
+	}
+
+	var playbookSpec v1.PlaybookSpec
+	if err := json.Unmarshal(run.Spec, &playbookSpec); err != nil {
+		return ctx.Oops().Wrap(err)
+	}
+
+	completedActions, err := run.GetActions(ctx.DB())
+	if err != nil {
+		return ctx.Oops().Wrap(err)
+	}
+
+	var aiActionNames, notificationActionNames []string
+	for _, action := range playbookSpec.Actions {
+		if action.AI != nil {
+			aiActionNames = append(aiActionNames, action.Name)
+		}
+		if action.Notification != nil {
+			notificationActionNames = append(notificationActionNames, action.Name)
+		}
+	}
+
+	sendHistoryUpdate := models.NotificationSendHistory{
+		ID: *run.NotificationSendID,
+	}
+	for _, action := range completedActions {
+		if action.Status != models.PlaybookActionStatusCompleted || action.Result == nil {
+			continue
+		}
+
+		if lo.Contains(aiActionNames, action.Name) {
+			if diagnosisReport, ok := action.Result["json"].(string); ok {
+				var aiDiagnosisReport map[string]string
+				if err := json.Unmarshal([]byte(diagnosisReport), &aiDiagnosisReport); err != nil {
+					return ctx.Oops().Wrap(err)
+				}
+
+				if headline, ok := aiDiagnosisReport["headline"]; ok {
+					sendHistoryUpdate.ResourceHealthDescription = headline
+				}
+			}
+		}
+
+		if lo.Contains(notificationActionNames, action.Name) {
+			if slackMsg, ok := action.Result["slack"].(string); ok {
+				sendHistoryUpdate.Body = &slackMsg
+			} else if body, ok := action.Result["body"].(string); ok {
+				sendHistoryUpdate.Body = &body
+			}
+		}
+	}
+
+	if len(sendHistoryUpdate.ResourceHealthDescription) > 0 {
```

Comment:
What happens if there is just a notification in the playbook?

Reply:
We save the body sent by the notification action to the send history.